### ISSUE TITLE
can't embed sub in format

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10758]
Migrated from [rt.perl.org#78550](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=78550) (status was 'resolved')

Searchable as RT78550$

**Comment by p5pRT** [comment]
### From zefram@fysh.org

#### Created by zefram@fysh.org



$ cat t0\.pl
format EE =
@&#8203;\<\<
\(sub \{ \}\, "xyz"\)
\.
$ perl t0\.pl
perl&#8203;: toke\.c&#8203;:13500&#8203;: Perl\_start\_subparse&#8203;: Assertion \`\(\(svtype\)\(\(\(my\_perl\->Icompcv\)\)\->sv\_flags & 0xff\)\) == SVt\_PVCV' failed\.

Assertion fails with a debugging Perl\, but a non\-debugging Perl \(which
doesn't check the assertion\) doesn't seem to have any problem with it\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.13.6:

Summary of my perl5 (revision 5 version 13 subversion 6) configuration:
  Derived from: 5f211341d665e81a03fbd4935e4e87f70d937fc3
  Platform:
    osname=linux, osvers=2.6.26-2-686, archname=i386-linux-thread-multi
    uname='linux vigo.rous.org 2.6.26-2-686 #1 smp mon aug 30 07:01:57 utc 2010 i686 gnulinux '
    config_args='-des -Darchname=i386-linux -Dcccdlflags=-fPIC -Dccdlflags=-rdynamic -Dprefix=/home/zefram/usr/perl/perl_install/perl-blead-i32-f52 -Dman1ext=1 -Dman3ext=3perl -Duselargefiles -Dusethreads -Uafs -Ud_csh -Uusesfio -Uusenm -Duseshrplib -Dusedevel -Ui_db -DDEBUGGING'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.3.2', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lgdbm -ldl -lm -lcrypt -lutil -lpthread -lc -lgdbm_compat
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.7.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.7'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-rpath,/home/zefram/usr/perl/perl_install/perl-blead-i32-f52/lib/5.13.6/i386-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -g -L/usr/local/lib -fstack-protector'


Characteristics of this binary (from libperl): 
  Compile-time options: DEBUGGING MULTIPLICITY PERL_DONT_CREATE_GVSV
                        PERL_IMPLICIT_CONTEXT PERL_MALLOC_WRAP
                        PERL_TRACK_MEMPOOL PERL_USE_DEVEL USE_ITHREADS
                        USE_LARGE_FILES USE_PERLIO USE_PERL_ATOF
                        USE_REENTRANT_API
  Locally applied patches:
	uncommitted-changes
  Built under linux
  Compiled at Oct 24 2010 14:09:57
  @INC:
    /home/zefram/usr/perl/pg/lib
    /home/zefram/usr/perl/perl_install/perl-blead-i32-f52/lib/site_perl/5.13.6/i386-linux-thread-multi
    /home/zefram/usr/perl/perl_install/perl-blead-i32-f52/lib/site_perl/5.13.6
    /home/zefram/usr/perl/perl_install/perl-blead-i32-f52/lib/5.13.6/i386-linux-thread-multi
    /home/zefram/usr/perl/perl_install/perl-blead-i32-f52/lib/5.13.6
    .

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed in 9a715434797\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
